src/csp/csp-auto-suggest.js

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import { Audit } from '@adobe/spacecat-shared-data-access';
+
+const AUDIT_TYPE = Audit.AUDIT_TYPES.SECURITY_CSP;
+const scriptRegExp = /<script.*?(?:\/>|<\/script>)/gs;
+const hasNonceRegExp = /<script[^>]*? nonce=/s;
+
+async function determineSuggestionsForPage(url, context, site) {
+  const { log } = context;
+
+  log.debug(`[${AUDIT_TYPE}] [Site: ${site.getId()}]: Fetching page '${url}' for CSP auto-suggest`);
+
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error(`Failed to fetch page from ${url}`);
+  }
+
+  const responseBody = await response.text();
+
+  if (!responseBody) {
+    log.warn(`[${AUDIT_TYPE}] [Site: ${site.getId()}] [Url: ${url}]: no script page content found`);
+    return null;
+  }
+
+  let suggestedBody = responseBody;
+  const findings = [];
+  const scriptTags = responseBody.matchAll(scriptRegExp);
```

Comment:
Instead of regex, you could use `cheerio` for getting the tags and attributes. The library is already installed in the project.